### PR TITLE
docs: Update doc to use `ng` firebase schematics for deployment

### LINKED ISSUE #35274: Update the deployment instructions in getting started
# 📚 Docs or angular.io bug report

### Description

With the tighter integration of `@angular/fire` with Angular CLI and the `ng deploy` command that we introduced, we can simplify the deployment section in the getting started [guide](https://angular.io/start/deployment#deployment).

### What's the affected URL?**

https://angular.io/start/deployment#deployment

### Expected vs Actual Behavior**

Expected:
```
ng add @angular/fire
ng deploy
```

Skip part of the steps [here](https://angular.io/start/deployment#hosting-an-angular-app-on-firebase).

### PR DESCRIPTION
**Pull request by sonukapoor**
Previously, the `deployment` section, was using the `firebase`
CLI to deploy the angular project into firebase. With the better
integration through the `fire` schematics, it is now easier to
deploy angular applications into firebase. This commit takes
care of this, by outlined the required steps for deployment.

Closes #35274

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #35274

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by sonukapoor** [comment]
@mgechev Take a look and let me know what you think.

File: aio/content/start/deployment.md [hunk 1]
```diff
@@ -66,13 +66,33 @@ One of the easiest ways to get your site live is to host it using Firebase.
 
 1. Sign up for a firebase account on [Firebase](https://firebase.google.com/ "Firebase web site").
 1. Create a new project, giving it any name you like.
-1. Install the `firebase-tools` CLI that will handle your deployment using `npm install -g firebase-tools`.
+1. Add the `@angular/fire@next` schematics that will handle your deployment using `ng add @angular/fire@next`.
```
Comment (IgorMinar) [inline aio/content/start/deployment.md]: We should not be encouraging the use of next pre-releases in our mainstream docs.

Do we need some improvements from the next version for these instructions to work? Of so, we should wait until they are released in the stable version.

File: aio/content/start/deployment.md [hunk 2]
```diff
@@ -66,13 +66,33 @@ One of the easiest ways to get your site live is to host it using Firebase.
...
 1. Connect your CLI to your Firebase account and initialize the connection to your project using `firebase login` and `firebase init`.
 1. Follow the prompts to select the `Firebase` project you are creating for hosting.
-  - Select the `Hosting` option on the first prompt.
-  - Select the project you previously created on Firebase.
-  - Select `dist/my-project-name` as the public directory.
-1. Deploy your application with `firebase deploy`, because the command `firebase init` has created a `firebase.json` file that tells Firebase how to serve your app.
+    - Select the `Hosting` option on the first prompt.
+    - Select the project you previously created on Firebase.
+    - Select `dist/my-project-name` as the public directory.
+1. Update your `angular.json` by inserting a `deploy` builder.
```
Comment (IgorMinar) [inline aio/content/start/deployment.md]: Is this really required? Doesn't the ng add schematic do this automatically?

**Review by IgorMinar — CHANGES_REQUESTED** [review]

File: aio/content/start/deployment.md (hunk 1 above)
Comment (sonukapoor) [reply aio/content/start/deployment.md]: The current version does not work properly. It throws errors when there is no `targets` property defined in the `.firebaserc`.

File: aio/content/start/deployment.md (hunk 2 above)
Comment (sonukapoor) [reply aio/content/start/deployment.md]: Yes, it is. The schematics do not do that. I would be happy to contribute to that if you point me into the right direction.

File: aio/content/start/deployment.md (hunk 1 above)
Comment (mgechev) [reply aio/content/start/deployment.md]: We can hold this PR off for a bit in this case. Firebase are about to release version 6 in the next a couple of days.

File: aio/content/start/deployment.md (hunk 2 above)
Comment (mgechev) [reply aio/content/start/deployment.md]: `ng add` in `@angular/fire` adds a deploy target automatically, this shouldn't be a requirement.

**Comment by mgechev** [comment]
@sonukapoor thanks for the PR! You should be able to address @IgorMinar's comments after the 6.0 release of `@angular/fire`.

**Comment by IgorMinar** [comment]
Sgtm. Thanks @mgechev

**Comment by sonukapoor** [comment]
@mgechev / @IgorMinar My bad. It seems that you need to have a firebase project set up first before you run `ng add @angular/fire`. I created a brand new project and used the current version of `5.4.2`. It updated the `angular.json` for me and everything else including `ng deploy` worked as expected. I have updated the doc. Please re-review. Thanks

File: aio/content/start/deployment.md [hunk 3]
```diff
@@ -66,13 +66,13 @@ One of the easiest ways to get your site live is to host it using Firebase.
...
-1. Install the `firebase-tools` CLI that will handle your deployment using `npm install -g firebase-tools`.
+1. Add the `@angular/fire` schematics that will handle your deployment using `ng add @angular/fire`.
 1. Connect your CLI to your Firebase account and initialize the connection to your project using `firebase login` and `firebase init`.
 1. Follow the prompts to select the `Firebase` project you are creating for hosting.
-  - Select the `Hosting` option on the first prompt.
-  - Select the project you previously created on Firebase.
-  - Select `dist/my-project-name` as the public directory.
-1. Deploy your application with `firebase deploy`, because the command `firebase init` has created a `firebase.json` file that tells Firebase how to serve your app.
+    - Select the `Hosting` option on the first prompt.
+    - Select the project you previously created on Firebase.
```
Comment (IgorMinar) [inline aio/content/start/deployment.md]: Is this extra whitespace needed? Seems wrong.

**Review by IgorMinar — CHANGES_REQUESTED** [review]
Thanks for following up on this. I did find it odd that the next version would be required but I haven't tested this myself.

File: aio/content/start/deployment.md (hunk 3 above)
Comment (sonukapoor) [reply aio/content/start/deployment.md]: It is. The space is used as an indentation. Here is how it currently looks:

image

Here is with the space using a markdown preview:

image

Comment (sonukapoor) [reply aio/content/start/deployment.md]: @IgorMinar Can you please add the `aio-preview` label, so that we can see the output?

Comment (sonukapoor) [reply aio/content/start/deployment.md]: Thanks @IgorMinar. How is this?

image

Comment (IgorMinar) [reply aio/content/start/deployment.md]: Lgtm. Thanks

**Review by IgorMinar — APPROVED** [review]
Lgtm now! Thank you

**Pull request closed without merging**